scene/animation/animation_mixer.cpp

```diff
@@ -1733,7 +1740,7 @@ void AnimationMixer::_blend_apply() {
 			case Animation::TYPE_VALUE: {
 				TrackCacheValue *t = static_cast<TrackCacheValue *>(track);
 
-				if (!t->is_variant_interpolatable || (callback_mode_discrete == ANIMATION_CALLBACK_MODE_DISCRETE_DOMINANT && t->use_discrete)) {
+				if (!t->is_variant_interpolatable || !t->use_continuous || (callback_mode_discrete == ANIMATION_CALLBACK_MODE_DISCRETE_DOMINANT && t->use_discrete)) {
```

Comment:
Is the reason t hat if use_continuous was not initialized to `callback_mode_discrete == ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS` here, then it will need to be

`(!!t->use_continuous && callback_mode_discrete != ANIMATION_CALLBACK_MODE_DISCRETE_FORCE_CONTINUOUS)`

but won't line 1430 have always run before hitting this?

Reply:
At the time of blending about line 1430, it is not yet possible to determine if discrete is present there or not. Once iteration is complete, it is necessary to determine if the result should really be applied.